proxy/tests/shutdown.rs

```diff
@@ -0,0 +1,144 @@
+mod support;
+use self::support::*;
+
+#[test]
+fn h2_goaways_connections() {
+    let _ = env_logger::try_init();
+
+    let (shdn, rx) = shutdown_signal();
+
+    let srv = server::http2().route("/", "hello").run();
+    let ctrl = controller::new().run();
+    let proxy = proxy::new()
+        .controller(ctrl)
+        .inbound(srv)
+        .shutdown_signal(rx)
+        .run();
+    let client = client::http2(proxy.inbound, "shutdown.test.svc.cluster.local");
+
+    assert_eq!(client.get("/"), "hello");
+
+    shdn.signal();
+
+    client.wait_for_closed();
+}
+
+#[test]
+fn h2_exercise_goaways_connections() {
```

Comment:
This is the unit test that gets 100 responses, but hasn't read the bodies yet (each 16kb), and thanks to flow control, we know a majority of the data is still on the server. After seeing the headers, then the shutdown signal is triggered, and then the bodies are streamed.

Removing the drain watcher from our `h2` connections in the proxy makes this test fail hard.

Reply:
Very minor, but it might be nice if that was described in a comment on the test rather than in the PR review?